Libraries/ToTripleSlashPorter.cs

```diff
@@ -0,0 +1,133 @@
+﻿#nullable enable
+using Libraries.Docs;
+using Libraries.RoslynTripleSlash;
+using Microsoft.Build.Logging;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.MSBuild;
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+
+namespace Libraries
+{
+    public class ToTripleSlashPorter
+    {
+        private readonly Configuration Config;
+        private readonly DocsCommentsContainer DocsComments;
+
+        public ToTripleSlashPorter(Configuration config)
+        {
+            if (config.Direction != Configuration.PortingDirection.ToTripleSlash)
+            {
+                throw new InvalidOperationException($"Unexpected porting direction: {config.Direction}");
+            }
+            Config = config;
+            DocsComments = new DocsCommentsContainer(config);
+        }
+
+        public void Start()
+        {
+            DocsComments.CollectFiles();
+            if (!DocsComments.Types.Any())
+            {
+                Log.ErrorAndExit("No Docs Type APIs found.");
+            }
+
+            Log.Info("Porting from Docs to triple slash...");
+
+            MSBuildWorkspace workspace;
+            try
+            {
+                workspace = MSBuildWorkspace.Create();
+            }
+            catch (ReflectionTypeLoadException)
+            {
+                Log.ErrorAndExit("The MSBuild directory was not found in PATH. Use '-MSBuild <directory>' to specify it.");
+                return;
+            }
+
+            BinaryLogger? binLogger = null;
+            if (Config.BinLogger)
+            {
+                binLogger = new BinaryLogger()
+                {
+                    Parameters = Path.Combine(Environment.CurrentDirectory, Config.BinLogPath),
+                    Verbosity = Microsoft.Build.Framework.LoggerVerbosity.Diagnostic,
+                    CollectProjectImports = BinaryLogger.ProjectImportsCollectionMode.Embed
+                };
+            }
+
+            Project? project = workspace.OpenProjectAsync(Config.CsProj!.FullName, msbuildLogger: binLogger).Result;
+            if (project == null)
+            {
+                Log.ErrorAndExit("Could not find a project.");
+                return;
+            }
+
+            Compilation? compilation = project.GetCompilationAsync().Result;
```

Comment:
We still need to figure out how to get all compilations so that we get all the locations of an API for the cases where there is no an OS agnostic file for that API. Anyway, that could be done when we hit that scenario.